uimaj-core/src/test/java/org/apache/uima/analysis_engine/impl/PearAnalysisEngineWrapperTest.java

```diff
@@ -1,148 +1,166 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.uima.analysis_engine.impl;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+
+import org.apache.uima.pear.tools.PackageBrowser;
+import org.apache.uima.pear.tools.PackageInstaller;
+import org.apache.uima.pear.util.FileUtil;
+import org.apache.uima.resource.PearSpecifier;
+import org.apache.uima.resource.impl.Parameter_impl;
+import org.apache.uima.resource.impl.PearSpecifier_impl;
+import org.apache.uima.resource.metadata.impl.NameValuePair_impl;
+import org.apache.uima.test.junit_extension.JUnitExtension;
+import org.junit.Assert;
+
+import junit.framework.TestCase;
+
+public class PearAnalysisEngineWrapperTest extends TestCase {
+
+  // This parameter is defined in the PEAR specifier file!
+  private final static String PARAMETER_NAME = "StringParam";
+
+  private final static String PARAMETER_VALUE = "test";
+
+  private final static String PARAMETER_VALUE_OVERRIDE = "testOverride";
+
+  private PearAnalysisEngineWrapper pearAnalysisEngineWrapper;
+
+  private File tempInstallDirectory = null;
+
+  private PackageBrowser installedPearPackage;
+
+  public PearAnalysisEngineWrapperTest(String arg0) {
+
+    super(arg0);
+  }
+
+  @Override
+  protected void setUp() throws Exception {
+
+    this.pearAnalysisEngineWrapper = new PearAnalysisEngineWrapper();
+
+    this.tempInstallDirectory = this.createInstallationDirectory();
+
+    this.installedPearPackage = this.installPearPackage();
+  }
+
+  @Override
+  protected void tearDown() throws Exception {
+    this.removeInstallationDirectory();
+  }
+
+  public void testInitializeWithOverride() throws Exception {
+
+    PearSpecifier_impl pearSpecifier = new PearSpecifier_impl();
+    pearSpecifier.setPearPath(this.installedPearPackage.getRootDirectory().toString());
+    pearSpecifier.setPearParameters(new NameValuePair_impl(PARAMETER_NAME, PARAMETER_VALUE_OVERRIDE));
+
+    boolean initialized = this.pearAnalysisEngineWrapper.initialize(pearSpecifier, new HashMap<>());
+
+    assertThat(initialized).isTrue().as("Pear was initialized");
+    assertThat(pearAnalysisEngineWrapper.getConfigParameterValue(PARAMETER_NAME))
+        .isEqualTo(PARAMETER_VALUE_OVERRIDE)
+        .as("The value of StringParam was overridden");
+  }
+
+  public void testInitializeWithOverrideLegacy() throws Exception {
+
+    PearSpecifier_impl pearSpecifier = new PearSpecifier_impl();
+    pearSpecifier.setPearPath(this.installedPearPackage.getRootDirectory().toString());
+    pearSpecifier.setParameters(new Parameter_impl(PARAMETER_NAME, PARAMETER_VALUE_OVERRIDE));
+
+    boolean initialized = this.pearAnalysisEngineWrapper.initialize(pearSpecifier, new HashMap<>());
+
+    assertThat(initialized).isTrue().as("Pear was initialized");
```

Comment:
thanks for the more readable test case syntax :-)

Reply:
You're welcome. I like assertj. I'm thinking of adding some UIMA-specific asserts either in Core or in uimaFIT.